backend/de.metas.material/cockpit/src/main/java/de/metas/material/cockpit/availableforsales/AvailableForSalesService.java

```diff
@@ -170,13 +172,21 @@ private void syncAvailableForSalesTable(@NonNull final AvailableForSalesQuery av
 
 		final Set<I_MD_Available_For_Sales> recordsToBeDeleted = new HashSet<>(availableForSalesRecords);
 		recordsToBeDeleted.removeAll(recordsToBeUpdated.keySet());
```

Comment:
eek, i hadn't seen this before. it makes me nervous because i don't really know how hashcode is implemented.
i'm fine if you can explain how this is safe..otherwise please use some way that works with Integers, RepoIdAwares (e.g. instances of `Map<Integer, I_MD_Available_For_Sales>`) where we can be sure that it works